frontend/src/index.pug

```diff
@@ -136,7 +136,7 @@ html
             .summary-charts__fileformat--breakdown(v-if="filterBreakdown")
               template(v-if="filterGroupSelection === 'groupByRepos'")
                 .summary-charts__fileformat--breakdown__legend(
-                    v-for="fileFormat in Object.keys(repo[0].fileFormatContribution)")
+                    v-for="fileFormat in getFileFormatsTouchedInRepo(repo)")
```

Comment:
Why we now have a situation where different users may get different file types?

Reply:
Because some users may have contributed to certain file formats, and some other may not. So right now, if all file formats are captured, this is how the `commits.json` file will look like:
![image](https://user-images.githubusercontent.com/35756209/60697852-f59c3080-9f1e-11e9-9eeb-30e0e39468ce.png)

Where file formats that are not touched by authors will no longer be assigned to the authors under the `authorFileFormatContributionMap` object.

Should we stick to this form? Or should we take an approach where:- 
1. Detect all file formats touched by all authors, store them in a list first
2. Then set all these detected formats to each authors, and default to 0 for those formats not touched by the authors
?

Reply:
OK，I think your approach is better. Let's just use this one.